spec/dsl_spec_helper.rb

```diff
@@ -3,72 +3,60 @@
 
 require "sorbet-runtime"
 require "minitest/spec"
-require "tapioca/helpers/test/content"
-require "tapioca/helpers/test/template"
-require "tapioca/helpers/test/isolation"
+require "tapioca/helpers/test/dsl_compiler"
 
 class DslSpec < Minitest::Spec
   extend T::Sig
-  include Kernel
-  include Tapioca::Helpers::Test::Content
-  include Tapioca::Helpers::Test::Template
-  include Tapioca::Helpers::Test::Isolation
+  include Tapioca::Helpers::Test::DslCompiler
 
-  sig { void }
-  def after_setup
+  before do
     # Require the file that the target class should be loaded from
-    require(T.unsafe(self).target_class_file)
+    require(self.class.target_class_file)
+    use_dsl_compiler(self.class.target_class)
+    @expecting_errors = false
   end
 
-  sig { void }
-  def teardown
-    super
-    T.unsafe(self).subject.errors.clear
+  after do
+    assert_empty(generated_errors) unless @expecting_errors
+    generated_errors.clear
```

Comment:
If I'm writing my own `after` in my test, doesn't that mean I can't assert anything about errors because the `generated_errors` will be cleared before my hook runs?

Reply:
You can write it in a `before_teardown` method. Alternatively, we can move this to an `after_teardown` method too, so that `after` blocks can still have access to the errors.